src/Cli/Microsoft.TemplateEngine.Cli/PostActionProcessors/AddJsonPropertyPostActionProcessor.cs

```diff
@@ -0,0 +1,191 @@
+{
+    internal class AddJsonPropertyPostActionProcessor : PostActionProcessorBase
+    {
+        private const string JsonFileNameArgument = "jsonFileName";
+        private const string ParentPropertyPathArgument = "parentPropertyPath";
+        private const string NewJsonPropertyNameArgument = "newJsonPropertyName";
+        private const string NewJsonPropertyValueArgument = "newJsonPropertyValue";
+
+        private static readonly JsonSerializerOptions SerializerOptions = new JsonSerializerOptions
+        {
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+            WriteIndented = true
+        };
+
+        private static readonly JsonDocumentOptions DeserializerOptions = new JsonDocumentOptions
+        {
+            AllowTrailingCommas = true,
+            CommentHandling = JsonCommentHandling.Skip
+        };
+
+        public override Guid Id => ActionProcessorId;
+
+        internal static Guid ActionProcessorId { get; } = new Guid("695A3659-EB40-4FF5-A6A6-C9C4E629FCB0");
+
+        protected override bool ProcessInternal(
+            IEngineEnvironmentSettings environment,
+            IPostAction action,
+            ICreationEffects creationEffects,
+            ICreationResult templateCreationResult,
+            string outputBasePath)
+        {
+            if (!action.Args.TryGetValue(JsonFileNameArgument, out string? jsonFileName))
+            {
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.PostAction_ModifyJson_Error_ArgumentNotConfigured, JsonFileNameArgument));
+                return false;
+            }
+
+            IReadOnlyList<string> jsonFiles = FindFilesInCurrentProjectOrSolutionFolder(environment.Host.FileSystem, outputBasePath, matchPattern: jsonFileName, maxAllowedAboveDirectories: 1);
+
+            if (jsonFiles.Count == 0)
+            {
+                Reporter.Error.WriteLine(LocalizableStrings.PostAction_ModifyJson_Error_NoJsonFile);
+                return false;
+            }
+
+            if (jsonFiles.Count > 1)
+            {
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.PostAction_ModifyJson_Error_MultipleJsonFiles, jsonFileName));
+                return false;
+            }
+
+            // If no ParentPropertyPath is specified, the new JSON property must be added to the root of the
+            // document.
+            action.Args.TryGetValue(ParentPropertyPathArgument, out string? parentProperty);
+            action.Args.TryGetValue(NewJsonPropertyNameArgument, out string? newJsonPropertyName);
+            action.Args.TryGetValue(NewJsonPropertyValueArgument, out string? newJsonPropertyValue);
+
+            if (newJsonPropertyName == null)
+            {
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.PostAction_ModifyJson_Error_ArgumentNotConfigured, NewJsonPropertyNameArgument));
+                return false;
+            }
+
+            if (newJsonPropertyValue == null)
```

Comment:
can we move these validation checks to a separate method in order to make this method easier to read?

Reply:
In most cases, I'm in favor of doing this, however, in this case, I 'm not sure if it will improve readability.
I've made an attempt though in https://github.com/dotnet/sdk/pull/29880/commits/cb9979f497ba683d25307fbcdd3c717ba94dafe3